jms/src/main/scala/akka/stream/alpakka/jms/JmsProducerMessage.scala

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.jms
+
+object JmsProducerMessage {
+
+  sealed trait Envelope[+M <: JmsMessage, +PassThrough] {
+    def passThrough: PassThrough
+  }
+
+  def message[M <: JmsMessage, PassThrough](message: M, passThrough: PassThrough): Envelope[M, PassThrough] =
+    Message(message, passThrough)
+
+  def passThroughMessage[M <: JmsMessage, PassThrough](passThrough: PassThrough): Envelope[M, PassThrough] =
+    PassThroughMessage(passThrough)
+
+  final case class Message[+M <: JmsMessage, +PassThrough](message: M, passThrough: PassThrough)
+      extends Envelope[M, PassThrough]
+
+  final case class PassThroughMessage[+M <: JmsMessage, +PassThrough](passThrough: PassThrough)
```

Comment:
We may want to document the use case for `PassThroughMessage` similar to the Scaladocs in the Kafka case, or document it in the main documentation. Otherwise it may not be clear when and why to use this class.

Reply:
Good point, I'll add a section in the docs explaining that this allows to ack messages consumed without producing, and without the need for fan-out flows.